flink-cdc-connect/flink-cdc-source-connectors/flink-connector-postgres-cdc/src/main/java/org/apache/flink/cdc/connectors/postgres/source/reader/PostgresSourceReader.java

```diff
@@ -133,6 +135,17 @@ public void notifyCheckpointComplete(long checkpointId) throws Exception {
         }
     }
 
+    @Override
+    protected void onSplitFinished(Map finishedSplitIds) {
+        super.onSplitFinished(finishedSplitIds);
+
+        if (this.sourceConfig.getStartupOptions().isSnapshotOnly()) {
+            PostgresDialect dialect = (PostgresDialect) this.dialect;
+            boolean removed = dialect.removeSlot(dialect.getSlotName());
+            LOG.info("Remove slot '{}' result is {}.", dialect.getSlotName(), removed);
+        }
+    }
+
```

Comment:
```suggestion
@Override
    protected void onSplitFinished(Map<String, SourceSplitState> finishedSplitIds) {
        super.onSplitFinished(finishedSplitIds);

        for (SourceSplitState splitState : finishedSplitIds.values()) {
            SourceSplitBase sourceSplit = splitState.toSourceSplit();
            if (sourceSplit.isStreamSplit()  ) {
                StreamSplit streamSplit = sourceSplit.asStreamSplit();
                if(streamSplit.getStartingOffset().isAtOrAfter(streamSplit.getEndingOffset())){
                    PostgresDialect dialect = (PostgresDialect) this.dialect;
                    boolean removed = dialect.removeSlot(dialect.getSlotName());
                    LOG.info("Remove slot '{}' result is {}.", dialect.getSlotName(), removed);
                }

            }
        }
    }

```

It seems that even a snapshot split is finished, the global slot name will be removed.